### ISSUE TITLE
SimpleCaptcha NPE with Java 17, 21

### ISSUE THREAD

**Issue by avernet** [orbeon/orbeon-forms#6215]
See below for the beginning of the stack trace and [exception.txt](https://github.com/orbeon/orbeon-forms/files/14454878/exception.txt) for the full exception. I was able to reproduce the exception, which the customer reports does not happen with Java 8. Additionally, the customer reports that the image produced on Java is much clearer, which may or may not be related to this exception.

```
java.lang.NullPointerException: Cannot load from int array because "this.inPixels" is null
        at com.jhlabs.image.TransformFilter.imageComplete(TransformFilter.java)
        at java.desktop/sun.awt.image.OffScreenImageSource.produce(OffScreenImageSource.java:193)
        at java.desktop/sun.awt.image.OffScreenImageSource.addConsumer(OffScreenImageSource.java:66)
        at java.desktop/sun.awt.image.OffScreenImageSource.startProduction(OffScreenImageSource.java:80)
        at java.desktop/java.awt.image.FilteredImageSource.startProduction(FilteredImageSource.java:184)
        at java.desktop/sun.awt.image.ImageRepresentation.startProduction(ImageRepresentation.java:728)
        at java.desktop/sun.awt.image.ToolkitImage.addWatcher(ToolkitImage.java:221)
        at java.desktop/sun.awt.image.ToolkitImage.getWidth(ToolkitImage.java:110)
        at java.desktop/sun.java2d.SunGraphics2D.drawImage(SunGraphics2D.java:3419)
        at nl.captcha.util.ImageUtil.applyFilter(ImageUtil.java:15)
        at nl.captcha.gimpy.RippleGimpyRenderer.gimp(RippleGimpyRenderer.java:31)
        at nl.captcha.Captcha$Builder.gimp(Captcha.java:180)
        at nl.captcha.Captcha$Builder.gimp(Captcha.java:171)
        at org.orbeon.xbl.SimpleCaptcha$.createCaptcha(SimpleCaptcha.scala:24)
 ```

[+1 from customer](https://3.basecamp.com/3600924/buckets/18345583/messages/7134080225)

**Comment by avernet** [comment]
The problem we're seeing was [reported](https://sourceforge.net/p/simplecaptcha/discussion/494129/thread/a1318d9a95/) in March 2023, but no one responded.

The latest version of SimpleCaptcha, 1.2.1, was released in 2011. Kaptcha, described as "a modern version of the SimpleCaptcha project," had its last commit in 2019 on https://github.com/penggle/kaptcha, which appears to be the most popular Kaptcha repository. Kaptcha might be a better solution, especially if replacing SimpleCaptcha with Kaptcha is straightforward.

**Comment by ebruchez** [comment]
- not great to upgrade to something abandoned for 5 years
- maybe we can compile and get SimpleCaptcha to work again

**Comment by avernet** [comment]
Erik suggests recompiling SimpleCaptcha to catch the exception.

**Comment by avernet** [comment]
For reference, this shipped in Orbeon Forms 2023.1.1 as well as 2022.1.7.
(What happened to those GitHub projects?)
